mux_test.go

```diff
@@ -288,6 +288,34 @@ func TestMux(t *testing.T) {
 
 	expectUpdate(persistence.InvoiceStateSettled)
 
+	cancelSubscription()
+
+	// Create another new invoice for overpayment tests
+	invoice, testPreimage, err = creator.Create(
+		20000, time.Minute, "test 3", nil, 40,
+	)
+	require.NoError(t, err)
+	require.NotNil(t, invoice)
+
+	testHash = testPreimage.Hash()
+
+	cancelSubscription, err = registry.Subscribe(testHash, func(update InvoiceUpdate) {
+		logger.Sugar().Infow("Payment received", "state", update.State)
+
+		updateChan <- update
+	})
+	require.NoError(t, err)
+
+	// Regenerate onion blob for new hash.
+	onionBlob = genOnion()
+
+	htlcChan1 <- receiveHtlc(30, 20001)
+
+	// Expect failure for MPP overpayment
+	// TODO(aakselrod): decrypt and check failure message
+	expectResponse(<-responseChan1, 30, routerrpc.ResolveHoldForwardAction_FAIL)
+	require.ErrorIs(t, registry.RequestSettle(testHash), ErrSetNoLongerComplete)
```

Comment:
Maybe something is not right. I'd expect this error only in the case that one or more htlcs of a set have been release (htlc release timeout), but there are still some left.

Overpayment should hit https://github.com/bottlepay/lnmux/blob/fd33bf0d8627d325586f7f844ed37e7cfe7b1615/invoiceregistry.go#L866 and I'd expect the set to be removed from memory.

Reply:
Yes, looked more closely at it and we definitely have a bug. The idea is that we only keep invoices in memory (`InvoiceRegistry.invoices`) that have a non-empty htlc set. But the check above and a few other in that same area leave an empty set.

Reply:
Did a bigger refactor to be more certain of the correctness: https://github.com/bottlepay/lnmux/pull/19

Reply:
With the refactor, I was able to make a better test case.